File: python/abc011_3.py
n=int(input())
l=[int(input()) for _ in range(3)]
l=set(l)
cnt=0
ans='YES'
if n in l: ans='NO'
while n>0:
	x=n-3
	while x in l or x<0:
		x+=1
	if x>n:
		ans='NO';break
	n=x;cnt+=1
	if cnt>100:
		ans='NO';break
print(ans)
		

File: cpp/abc011_3.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
	int n;
	cin>>n;
	vector<int> ns(3);
	for(int i=0;i<3;i++)cin>>ns[i];
	int cnt=100;
	while(cnt--){
		n-=3;
		bool flg=true;
		while(flg){
			flg=true;
			for(int i=0;i<3;i++)if(ns[i]==n){
				flg=false;break;
			}
			n++;
		}
	}
	
	cout<<((n<=0)?"YES":"NO")<<endl;
}